File: Samples-Xamarin/Adapters/ExamplesAdapter.cs
using System;
using Android.Widget;
using Java.Util;
using Android.Views;

namespace Samples
{
	public class ExamplesAdapter : BaseExpandableListAdapter {

		private Java.Lang.Object[] keys;

		private HashMap source;

		public ExamplesAdapter(HashMap source){
			this.source = source;

			Java.Lang.Object[] k = new Java.Lang.Object[source.Size ()];
			int counter = 0;
			foreach (Java.Lang.Object o in this.source.KeySet()) {
				k [counter++] = o;
			}

			this.keys = k;
		}

		public override int GroupCount {
			get {
				return this.keys.Length;
			}
		}


		public override int GetChildrenCount(int groupPosition) {
			return ((ArrayList)this.source.Get(this.keys[groupPosition])).Size();
		}


		public override Java.Lang.Object GetGroup(int groupPosition) {
			return this.keys[groupPosition];
		}


		public override Java.Lang.Object GetChild(int groupPosition, int childPosition) {
			return ((ArrayList)this.source.Get(this.keys[groupPosition])).Get(childPosition);
		}


		public override long GetGroupId(int groupPosition) {
			return this.keys[groupPosition].GetHashCode();
		}


		public override long GetChildId(int groupPosition, int childPosition) {
			return ((ArrayList)this.source.Get(this.keys[groupPosition])).Get(childPosition).GetHashCode();
		}


		public override bool HasStableIds {
			get {
				return true;
			}
		}


		public override View GetGroupView(int groupPosition, bool isExpanded, View convertView, ViewGroup parent) {
			View container = View.Inflate(parent.Context, Resource.Layout.list_item_group, null);
			TextView txtGroupName = (TextView)container.FindViewById(Resource.Id.txtFeatureName);
			txtGroupName.Text = this.GetGroup(groupPosition).ToString();
			return container;
		}


		public override View GetChildView(int groupPosition, int childPosition, bool isLastChild, View convertView, ViewGroup parent) {
			View container = View.Inflate(parent.Context, Resource.Layout.list_item_child, null);
			TextView txtGroupName = (TextView)container.FindViewById(Resource.Id.txtExampleName);
			ExampleFragment child = (ExampleFragment)this.GetChild(groupPosition, childPosition);
			txtGroupName.Text = child.Title();
			return container;
		}


		public override bool IsChildSelectable(int groupPosition, int childPosition) {
			return true;
		}
	}
}

File: Samples-Java/Samples/src/main/java/activities/ExamplesAdapter.java
package activities;

import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseExpandableListAdapter;
import android.widget.TextView;

import com.telerik.android.sdk.R;

import java.util.ArrayList;
import java.util.LinkedHashMap;

public class ExamplesAdapter extends BaseExpandableListAdapter {

    private Object[] keys;

    private LinkedHashMap<String, ArrayList<ExampleFragment>> source;

    public ExamplesAdapter(LinkedHashMap<String, ArrayList<ExampleFragment>> source){
        this.source = source;
        this.keys = this.source.keySet().toArray();
    }

    @Override
    public int getGroupCount() {
        return this.source.keySet().size();
    }

    @Override
    public int getChildrenCount(int groupPosition) {
        return this.source.get(this.keys[groupPosition]).size();
    }

    @Override
    public Object getGroup(int groupPosition) {
        return this.keys[groupPosition];
    }

    @Override
    public Object getChild(int groupPosition, int childPosition) {
        return this.source.get(this.keys[groupPosition]).get(childPosition);
    }

    @Override
    public long getGroupId(int groupPosition) {
        return this.keys[groupPosition].hashCode();
    }

    @Override
    public long getChildId(int groupPosition, int childPosition) {
        return this.source.get(this.keys[groupPosition]).get(childPosition).hashCode();
    }

    @Override
    public boolean hasStableIds() {
        return true;
    }

    @Override
    public View getGroupView(int groupPosition, boolean isExpanded, View convertView, ViewGroup parent) {
        View container = View.inflate(parent.getContext(), R.layout.list_item_group, null);
        TextView txtGroupName = (TextView)container.findViewById(R.id.txtFeatureName);
        txtGroupName.setText(String.valueOf(this.getGroup(groupPosition)));
        return container;
    }

    @Override
    public View getChildView(int groupPosition, int childPosition, boolean isLastChild, View convertView, ViewGroup parent) {
        View container = View.inflate(parent.getContext(), R.layout.list_item_child, null);
        TextView txtGroupName = (TextView)container.findViewById(R.id.txtExampleName);
        ExampleFragment child = (ExampleFragment)this.getChild(groupPosition, childPosition);
        txtGroupName.setText(child.title());
        return container;
    }

    @Override
    public boolean isChildSelectable(int groupPosition, int childPosition) {
        return true;
    }
}
